planning/behavior_path_planner/src/debug_utilities.cpp

```diff
@@ -0,0 +1,292 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "behavior_path_planner/debug_utilities.hpp"
+
+namespace marker_utils
+{
+using behavior_path_planner::ShiftPoint;
+using behavior_path_planner::util::calcPathArcLengthArray;
+using behavior_path_planner::util::shiftPose;
+using std_msgs::msg::ColorRGBA;
+using tier4_autoware_utils::createMarkerColor;
+using tier4_autoware_utils::createMarkerOrientation;
+using tier4_autoware_utils::createMarkerScale;
+using tier4_autoware_utils::createPoint;
+using visualization_msgs::msg::Marker;
+
+MarkerArray createPoseMarkerArray(
+  const Pose & pose, std::string && ns, const int32_t & id, const float & r, const float & g,
+  const float & b)
+{
+  MarkerArray msg;
+
+  Marker marker = createDefaultMarker(
+    "map", rclcpp::Clock{RCL_ROS_TIME}.now(), std::move(ns), id, Marker::ARROW,
+    createMarkerScale(0.7, 0.3, 0.3), createMarkerColor(r, g, b, 0.999));
+  marker.pose = pose;
+  msg.markers.push_back(marker);
+
+  return msg;
+}
+
+MarkerArray createPathMarkerArray(
+  const PathWithLaneId & path, std::string && ns, const int64_t & lane_id, const float & r,
+  const float & g, const float & b)
+{
+  const auto arclength = calcPathArcLengthArray(path);
+  const auto current_time = rclcpp::Clock{RCL_ROS_TIME}.now();
+  MarkerArray msg;
+  const int32_t uid = bitShift(lane_id);
+  int32_t i{0};
+  int32_t idx{0};
+  for (const auto & p : path.points) {
+    Marker marker = createDefaultMarker(
```

Comment:
@zulfaqar-azmi-t4 
these lines except pose is constant value in every loop so please add these fileds before loop
```
for (const auto & p : path.points) {
    Marker marker{};
    marker.header.frame_id = "map";
    marker.header.stamp = current_time;
    marker.ns = ns;
    marker.id = uid + i++;
    marker.lifetime = rclcpp::Duration::from_seconds(0.2);
    marker.type = Marker::ARROW;
    marker.action = Marker::ADD;
    marker.pose = p.point.pose;
    marker.scale = tier4_autoware_utils::createMarkerScale(0.2, 0.1, 0.3);
    marker.color = tier4_autoware_utils::createMarkerColor(r, g, b, 0.999);
    msg.markers.push_back(marker);

```

Reply:
@taikitanaka3 
I wonder since it is constant at each loop, how about we do this instead?
```
Marker marker = createDefaultMarker("map", current_time, ns, 0, Marker::ARROW, createMarkerScale(0.2, 0.1, 0.3), createMarkerColor(r, g, b, 0.999));

Marker marker_text = createDefaultMarker("map", current_time, ns, 0, Marker::TEXT_VIEW_FACING, createMarkerScale(0.2, 0.1, 0.3), createMarkerColor(1, 1, 1, 0.999));

for (const auto & p : path.points) {
    marker.id = uid + i++;
    marker.pose = p.point.pose;
    msg.markers.push_back(marker);
    if (idx % 10 == 0) {
      marker_text.id = uid + i++;
      std::stringstream ss;
      ss << std::fixed << std::setprecision(1) << "i=" << idx << "\ns=" << arclength.at(idx);
      marker_text.text = ss.str();
      msg.markers.push_back(marker_text);
    }
    ++idx;
  }
```
We don't need to construct the Marker each time and we push_back will make a copy of whatever current value marker has.

Reply:
yes this is much better I think

Reply:
done